website/src/components/CustomCarousel.tsx

```diff
@@ -15,20 +15,23 @@ interface CarouselProps {
     previousItem?: CarouselItem
     currentItemIndex?: number
     autoAdvance?: boolean
+    hideCarouselNav?: boolean
```

Comment:
Honestly, we may need to break this carousel into two when we port. There are enough deviations of behavior between designs of the "same" component for this to already feel incredibly brittle.

Reply:
Yeah I like that idea! Given how this component has evolved, I'd suggest we have a Vertical and a Horizontal carousel component and collocate some related logic into sibling files.

I like the rule of thumb of keeping tightly knit/scoped logic close by in sibling files and then if we have shared logic that get used in other components, then we break that logic out into the higher level directories like the higher level shared hooks folder.

So something like this seems more convenient:
```
components/
  Carousel/
    Horizontal.tsx
    Vertical.tsx
    index.tsx
    move.ts (the carousel hook)
    ...whatever else we need or could clean up into another sibling file
```

Reply:
Yeah, that sounds good to me! I can port over all the pages with carousels in one [issue](https://github.com/sourcegraph/about-replatform/issues/39) in the replatform and that will be a good chance for a rewrite